src/features/user/Widget/DonationLink/DonationLinkForm.tsx

```diff
@@ -0,0 +1,244 @@
+import {
+  ReactElement,
+  useContext,
+  useState,
+  useCallback,
+  useEffect,
+} from 'react';
+import {
+  Alert,
+  Autocomplete,
+  Button,
+  Snackbar,
+  styled,
+  Switch,
+  TextField,
+} from '@mui/material';
+import i18next from '../../../../../i18n';
+import { ProjectPropsContext } from '../../../../features/common/Layout/ProjectPropsContext';
+import AutoCompleteCountry from '../../../common/InputTypes/AutoCompleteCountryNew';
+import { UserPropsContext } from '../../../common/Layout/UserPropsContext';
+import InlineFormDisplayGroup from './InlineFormDisplayGroup';
+import supportedLanguages from '../../../../utils/language/supportedLanguages.json';
+import React from 'react';
+import ProjectSelectAutocomplete from '../../BulkCodes/components/ProjectSelectAutocomplete';
+import { Project } from '../../../common/Layout/BulkCodeContext';
+import { TENANT_ID } from '../../../../utils/constants/environment';
+const { useTranslation } = i18next;
+import styles from './DonationLinkForm.module.scss';
+import CopyToClipboard from '../../../common/CopyToClipboard';
+import ProjectsList from '../../../projects/screens/Projects';
+import {
+  MuiAutocomplete,
+  StyledAutoCompleteOption,
+} from '../../../common/InputTypes/AutoCompleteTheme';
+
+// TODOO - refactor code for reuse?
+const StyledForm = styled('form')((/* { theme } */) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: 24,
+  alignItems: 'flex-start',
+  '& .formButton': {
+    marginTop: 24,
+  },
+  '& .inputContainer': {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: 24,
+    width: '100%',
+  },
+}));
+
+interface DonationLinkFormProps {
+  projectsList: Project[] | null;
+}
+
+const DonationLinkForm = ({
+  projectsList,
+}: DonationLinkFormProps): ReactElement | null => {
+  const { user, contextLoaded } = useContext(UserPropsContext);
+  const [country, setCountry] = useState(
+    contextLoaded ? user.country : undefined
+  );
+  const [Languages, setLanguage] = useState({
+    langCode: 'en',
+    languageName: 'English',
+  });
+  const [donationUrl, setDonationUrl] = useState<string>('');
+  const { t, ready } = useTranslation(['donationLink', 'donate']);
+  const [localProject, setLocalProject] = useState<Project | null>(null);
+  const [isSupport, setIsSupport] = useState<boolean>(!user.isPrivate);
+  const [isCopied, setIsCopied] = useState<boolean>(false);
+  const [isTesting, setIsTesting] = useState<boolean>(false);
+  const [isProjectSelected, setIsProjectSelected] = useState<boolean>(false);
+  // Load all projects
+
+  const handleChange = () => {
+    const link = isTesting
+      ? 'http://paydev.pp.eco/'
+      : process.env.NEXT_PUBLIC_DONATION_URL;
+    const url = `${link}?country=${country}&locale=${Languages.langCode}${
+      localProject == null
+        ? ''
+        : `&to=${localProject.name.split(/\W+/).join('-').toLowerCase()}`
```

Comment:
@prachigarg19 @mohitb35 instead of splitting the project name you need to use the project id or slug.

Reply:
eg: yucatan or proj_9YZxwGeIkwALyW2k3MQ15Fh1